subprojects/file-collections/src/main/java/org/gradle/api/internal/file/FileCollectionListener.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.file;
+
+import org.gradle.api.file.FileCollection;
+import org.gradle.internal.service.scopes.EventScope;
+import org.gradle.internal.service.scopes.Scopes;
+
+@EventScope(Scopes.Build.class)
+public interface FileCollectionListener {
```

Comment:
Similar with `FileResourceListener`. 
Event scope of this listener is subject to discuss:
Eventually this listener should be passed to `DefaultFileCollectionFactory`. Latter has `@ServiceScope(Scope.Global.class)`. And this is a conflict of scopes